torch/_decomp/decompositions.py

```diff
@@ -1559,6 +1559,32 @@ def maybe_mask(vals, length, range_max, adaptive, dim):
     return ret / (length_h * length_w)
 
 
+@register_decomposition(aten.index_add_)
+def index_add_(
+    x: TensorLike,
+    dim: int,
+    index: TensorLike,
+    tensor: TensorLike,
+    *,
+    alpha: NumberType = 1,
+):
+    dim = utils.canonicalize_dims(x.ndim, dim)
+    utils.check(
+        index.ndim <= 1,
+        lambda: f"Index should have dimension 1 or 0 (got {index.ndim})",
+    )
+    if alpha != 1:
+        python_type = utils.dtype_to_type(x.dtype)
+        utils.check(
+            utils.is_weakly_lesser_type(type(alpha), python_type),
+            lambda: f"alpha argument of type {type(alpha)} cannot be safely cast to type {python_type}!",
+        )
+        tensor = torch._prims.mul(tensor, alpha)
```

Comment:
it's better to use `tensor*alpha` here, inductor doesn't have lowering for prims.mul (in general, use `torch.* ` ops in decomps, not prims.

Reply:
I'll change this in my next PR

Reply:
FWIW, here I was replicating the behaviour in `add`.